examples/multiclass/plot_multiclass_overview.py

```diff
@@ -0,0 +1,198 @@
+#   one class and the rest of the classes.
+# * :class:`~sklearn.multiclass.OutputCodeClassifier`: trains a set of binary
+#   classifiers where each classifier is trained to distinguish between
+#   a set of classes from the rest of the classes. The set of classes is
+#   defined by a codebook, which is randomly generated in scikit-learn. This
+#   method exposes a parameter `code_size` to control the size of the codebook.
+#   We set it above one since we are not interested in compressing the class
+#   representation.
+import pandas as pd
+from sklearn.tree import DecisionTreeClassifier
+from sklearn.multiclass import (
+    OneVsOneClassifier,
+    OneVsRestClassifier,
+    OutputCodeClassifier,
+)
+from sklearn.model_selection import cross_validate, RepeatedStratifiedKFold
+
+cv = RepeatedStratifiedKFold(n_splits=5, n_repeats=10, random_state=0)
+
+tree = DecisionTreeClassifier(random_state=0)
+ovo_tree = OneVsOneClassifier(tree)
+ovr_tree = OneVsRestClassifier(tree)
+ecoc = OutputCodeClassifier(tree, code_size=2)
+
+cv_results_tree = cross_validate(tree, X, y, cv=cv, n_jobs=2)
+cv_results_ovo = cross_validate(ovo_tree, X, y, cv=cv, n_jobs=2)
+cv_results_ovr = cross_validate(ovr_tree, X, y, cv=cv, n_jobs=2)
+cv_results_ecoc = cross_validate(ecoc, X, y, cv=cv, n_jobs=2)
+
+# %%
+# We can now compare the statistical performance of the different strategies.
+# We plot the score distribution of the different strategies.
+from matplotlib import pyplot as plt
+
+scores = pd.DataFrame(
+    {
+        "DecisionTreeClassifier": cv_results_tree["test_score"],
+        "OneVsOneClassifier": cv_results_ovo["test_score"],
+        "OneVsRestClassifier": cv_results_ovr["test_score"],
+        "OutputCodeClassifier": cv_results_ecoc["test_score"],
+    }
+)
+ax = scores.plot.kde(legend=True)
+ax.set_xlabel("Accuracy score")
+ax.set_xlim([0, 0.7])
+_ = ax.set_title(
+    "Density of the accuracy scores for the different multiclass strategies"
+)
+
+# %%
+# At a first glance, we can see that the built-in strategy of the decision
+# tree classifier is working quite well. One-vs-one and the error-correcting
+# output code strategies are working even better. However, the
+# one-vs-rest strategy is not working as well as the other strategies.
+#
+# Indeed, these results reproduce something reported in the literature
+# as in [2]_. However, the story is not as simple as it seems.
+#
+# The importance of hyperparameters search
+# ----------------------------------------
+#
+# It was later shown in [3]_ that the multiclass strategies would show similar
+# scores if the hyperparameters of the base classifiers are first optimized.
+#
+# Here we try to reproduce such result by at least optimizing the depth of the
+# base decision tree.
+from sklearn.model_selection import GridSearchCV
+
+param_grid = {"max_depth": [3, 5, 8]}
+tree_optimized = GridSearchCV(tree, param_grid=param_grid, cv=3, n_jobs=3)
+ovo_tree = OneVsOneClassifier(tree_optimized)
```

Comment:
What was the reason having the search "on the inside"? Concretely:

```python
ovo_tree = GridSearchCV(OneVsOneClassifier(tree), ...)
```

Reply:
I see the following points:

- The param grid directly define `max_depth` and not `estimator__max_depth`. I find it more intuitive.
- It avoids to wrap each meta-estimator in a grid-search.

Do you see any drawback of doing this approach?

Reply:
I'm trying to see if there is a "more correct" way to do the search. `GridSearchCV(OneVsOneClassifier(tree))` forces all the trees to have the same `max_depth`. This is analogous to running GridSearch on any scikit-learn classifier (which all natively support multi-class).

On the other hand, `OneVsOneClassifier(grid)` allows for each classifier to have different `max_depths`.

In any case, I think we can leave the PR as is.